public/js/index.js

```diff
@@ -1686,19 +1693,47 @@ function initDashboardSearch() {
                 this.tab = t;
             },
 
+            changeReposFilter: function(filter) {
+                this.reposFilter = filter;
+            },
+
+            showRepo: function(repo, filter) {
+                switch (filter) {
+                    case 'sources':
+                        return repo.owner.id == this.uid && !repo.mirror && !repo.fork;
+                    case 'forks':
+                        return repo.owner.id == this.uid && !repo.mirror && repo.fork;
+                    case 'collaborative':
+                        return repo.owner.id != this.uid;
+                    default:
+                        return true;
+                }
+            },
+
             searchKeyUp: function() {
                 this.searchRepos();
             },
 
             searchRepos: function() {
                 var self = this;
-                $.getJSON(this.searchURL(), function(result) {
-                    self.repos = result.data;
+                this.isLoading = true;
+                let searchedQuery = this.searchQuery;
```

Comment:
`let` -> `var` to not break on older browsers, since this is not being preprocessed.